backend/controllers/specializationsControllers.js

```diff
@@ -0,0 +1,149 @@
+const SpecializationsModel = require("../models/SpecializationsModel");
+const User = require("../models/UserModel");
+var mongoose = require("mongoose");
+
+module.exports.add = async (req, res) => {
+  try {
+    const { name, contributed_by, doctor_ids } = req.body;
+    if (!name) {
+      return res.status(400).json({ error: "Name is empty" });
+    }
+    const isValidContributed_by =
+      mongoose.Types.ObjectId.isValid(contributed_by); //true
+    if (!isValidContributed_by) {
+      return res.status(400).json({ error: "Invalid Contributor ID" });
+    }
+
+    // Validate user IDs
+    const contributor = await User.find({
```

Comment:
It is okay, we can also add specialization at the time of update profile or from profile page, but not make this mandatory

Reply:
Okay, i have commented the code for now.